### ISSUE TITLE
docker-compose pull instead of up

### ISSUE THREAD

**Issue by lindsve** [leighmcculloch/vagrant-docker-compose#44]
Are there any way to configure this so it only pulls the images and not starting the containers?

**Comment by leighmcculloch** [comment]
Not at the moment, but we could support your workflow. How does pulling fit into your workflow?

In the meantime what you can do is reduce your config to something like:

```ruby
config.vm.provision :docker
config.vm.provision :docker_compose
```

If you don't specify the `.yml` file it won't run `up`, it'll just install `docker-compose`.

Then you can add a shell provisioner that does the pull.

Example:

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/trusty64"

  config.vm.provision :docker
  config.vm.provision :docker_compose
  config.vm.provision :shell, inline: "cd /vagrant && docker-compose pull"
end
```

---

At the moment this plugin only runs `up` and not `pull`. It wouldn't be too difficult to add new boolean parameters that indicate which commands should be run, then you'd have full control over which ones run. I'll open a PR unless you want to take a stab at it? I'm thinking the config would look like this, with the new `commands` parameter:

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/trusty64"

  config.vm.provision :docker
  config.vm.provision :docker_compose,
    yml: "/vagrant/docker-compose.yml",
    options: "--x-networking",
    commands: { rm: true, build: true, pull: true, up: true },
    command_options: { rm: "", up: "-d --timeout 20"},
    run: "always"
end
```

**Comment by lindsve** [comment]
Sorry for the delayed response,

tl;dr, just close this issue for now.

We was looking into having two VMs (A/B) in parallel where we wanted to make sure that we could provision A before stopping B and then start the containers in A. The runtime environment where we will deploy our VMs will most likely have an unstable Internet connection so we wanted to make sure that all images was properly pulled. We have looked more into this and we should be able to solve this with proper tagging of the images instead

**Comment by weynhamz** [comment]
I already have the images pre-built and pushed to a custom registry, and I don't want to rebuild the images again when up the vagrant, so a 'pull' command before the 'up' would be nice.

**Comment by leighmcculloch** [comment]
I'm going to reopen because I think we can make this a feature too.

**Comment by gabz75** [comment]
Any update on this? Is there an alternative solution while waiting for this feature?

**Comment by leighmcculloch** [comment]
@gabz75 Would you (or anyone else) be up for taking a stab at this?
